### PR TITLE
[chore](docker) Add docker dev image with ldb-toolchain

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Add docker images `apache/incubator-doris:build-env-ldb-toolchain-latest`,
which is built with ldb-toolchain

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (Yes)
4. Does it need to update dependencies: (Yes)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -18,67 +18,55 @@
...
+RUN yum install -y byacc patch automake libtool make which file ncurses-devel gettext-devel unzip bzip2 zip util-linux wget git python2 \
+    java-11-openjdk-devel java-1.8.0-openjdk-devel
+
+# install ninjia
+RUN wget -q -O /tmp/ninja-linux.zip https://doris-thirdparty-repo.bj.bcebos.com/thirdparty/ninja-linux.zip \
+    && unzip /tmp/ninja-linux.zip -d /usr/bin/ \
+    && rm /tmp/ninja-linux.zip
+
+# install autoconf-2.69
+RUN wget https://doris-thirdparty-repo.bj.bcebos.com/thirdparty/autoconf-2.69.tar.gz && \
```
Comment (amosbird) [inline docker/Dockerfile:30]: If the image is based on centos 7, the system's autoconf and bison can be used directly with `yum install bison`, (autoconf should already be installed as a dependency of other packages, likely automake)

File: docker/Dockerfile.gcc10 [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    BASH_ENV="/opt/rh/devtoolset-10/enable" \
+    ENV="/opt/rh/devtoolset-10/enable"  \
+    PROMPT_COMMAND=". /opt/rh/devtoolset-10/enable" \
+    REPOSITORY_URL="https://doris-thirdparty-repo.bj.bcebos.com/thirdparty" \
+    DEFAULT_DIR="/var/local" \
+    PATH="/var/local/thirdparty/installed/bin:$PATH" \
+    DORIS_THIRDPARTY="/var/local/thirdparty"
+WORKDIR /root
+
+CMD ["/bin/bash"]
```
Comment (amosbird) [inline docker/Dockerfile.gcc10:123]: Missing carriage return here. Btw, why do we need this file?

Comment (morningman) [reply docker/Dockerfile.gcc10:123]: carriage return added.

> Btw, why do we need this file?

Actually not. I just rename the origin Dockerfile to Dockerfile.gcc10, for backup. Maybe delete it later.

File: docker/Dockerfile (hunk 1 above)
Comment (morningman) [reply docker/Dockerfile:30]: Remove

File: docker/Dockerfile [hunk 3]
```diff
@@ -18,67 +18,39 @@
...
-    rm /tmp/cmake-install.sh
-
-# build  ninja
-ARG NINJA_VER=1.10.2
-ARG NINJA_BASE_URL=https://github.com/ninja-build/ninja/releases/download/v${NINJA_VER}
-RUN wget -q ${NINJA_BASE_URL}/ninja-linux.zip -O /tmp/ninja-linux.zip && \
-    unzip /tmp/ninja-linux.zip -d /usr/bin/ && \
-    rm /tmp/ninja-linux.zip
+# install ninjia
+RUN wget -q -O /tmp/ninja-linux.zip https://doris-thirdparty-repo.bj.bcebos.com/thirdparty/ninja-linux.zip \
```
Comment (amosbird) [inline docker/Dockerfile:25]: I guess we can remove this too as ldb toolchain also contains ninja.

Comment (morningman) [reply docker/Dockerfile]: Done, I miss that

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**